code/datums/datum_click_handlers.dm

```diff
@@ -0,0 +1,125 @@
+/datum/click_handler
+//	var/mob_type
+	var/species
+	var/handler_name
+	var/one_use_flag = 1//drop client.CH after succes ability use
+	var/client/owner
+	var/icon/mouse_icon
+
+/datum/click_handler/New(client/_owner)
+	owner = _owner
+	if (mouse_icon)
+		owner.mouse_pointer_icon = mouse_icon
+
+//datum/click_handler/Prepare(/client/_owner)
+
+/datum/click_handler/Destroy()
+	..()
+	if (owner)
+		owner.CH = null
+		owner.mouse_pointer_icon=initial(owner.mouse_pointer_icon)
+	return ..()
+//	owner = null
+
+/datum/click_handler/proc/mob_check(mob/living/carbon/human/user) //Check can mob use a ability
+	return
+
+/datum/click_handler/proc/use_ability(mob/living/carbon/human/user,atom/target)
+	return
+
+/datum/click_handler/human/mob_check(mob/living/carbon/human/user)
+	if(ishuman(user))
+		if(user.species.name == src.species)
+			return 1
+	return 0
+
+/datum/click_handler/human/use_ability(mob/living/carbon/human/user,atom/target)
+	return
+
+//Changeling CH
+
+/datum/click_handler/changeling
+	mouse_icon = icon ('icons/changeling_mouse_icons.dmi')
+
+/datum/click_handler/changeling/mob_check(mob/living/carbon/human/user)
+	if(ishuman(user) && user.mind && user.mind.changeling)
+		return 1
+	return 0
+
+/datum/click_handler/changeling/use_ability(mob/living/carbon/human/user,atom/target) //Check can mob use a ability
+	if (user.stat == DEAD)
+		user << "No! You dead!"
```

Comment:
This could do with better grammar, something like
user << SPAN_WARNING("You must be alive to use abilities!")